src/utils.jl

```diff
@@ -1,5 +1,36 @@
-using Polynomials4ML: natural_indices
+using Polynomials4ML: natural_indices, ScalarPoly4MLBasis, lux
+using LuxCore: AbstractExplicitLayer
 
+struct Radial_basis
+   Rnl::AbstractExplicitLayer
+   Radialspec::Vector{NamedTuple}
+end
+
+Radial_basis(Rnl::AbstractExplicitLayer, spec_Rnl::Union{Vector{Int}, UnitRange{Int64}}) = 
+         Radial_basis(Rnl, [(n = i, ) for i in spec_Rnl])
+
+Radial_basis(Rnl::AbstractExplicitLayer) = 
+         try 
+            Radial_basis(Rnl,natural_indices(Rnl.basis)) 
+         catch 
+            error("The specification of this Radial_basis should be given explicitly!")
+         end
+
+# it is in its current form just for the purpose of testing
+function simple_radial_basis(basis::ScalarPoly4MLBasis,f_cut::Function=identity,f_trans::Function=identity; spec = nothing)
+   if isnothing(spec)
+      try 
+         spec = natural_indices(basis)
+      catch 
+         error("The specification of this Radial_basis should be given explicitly!")
+      end
+   end
+   
+   f(r) = f_cut(r) * f_trans(r)
+   
+   return Radial_basis(Chain(trans = WrappedFunction(xx -> [f(norm(x)) for x in xx]), 
```

Comment:
I think the usually `f_cut` and `f_trans` in ACE should be 

```julia
trans = WrappedFunction(xx -> (rr = norm.(xx); [f_cut(r)*f_trans(r) for r in rr]))
``` 
? 

From the current definition if `f_cut` and `f_trans` are identity the returned thing is `norm.(x).^2`

Reply:
Here we have removed the first l_xnx layer (or says absorbed it to the embedding) so the input is a set of positions rather than their norm. Hence it should be `norm(r)` but not `r`. One more thing regarding readability - do you think it is better to have two separated (sub-)layers for f_trans & f_cut respectively or is it better to have just one layer like what we wrote here - the reason why I am asking is that they are eventually Wrapped functions, so no information can be seen. But if we have one layer for each, we can at least name them as `cutoff` and `trans...` so that the user knows what to expect on those two invisible Wrapper functions.

Also, I agree that `f_cut = f_tran = identity` is nonsense. They are only in the current form for the sake of testing, as noted aside. The only requirement for them is just that they are both functions of `norm(x)`. For more elegant trans & cutoff functions, please refer to `forces.jl` - maybe we can consider migrating them to `radial.jl` and set them as default?